RecoLocalCalo/HcalRecAlgos/src/MahiFit.cc

```diff
@@ -69,16 +69,17 @@ void MahiFit::phase1Apply(const HBHEChannelInfo& channelData,
     auto const pedWidth = channelData.tsPedestalWidth(iTS);
 
     //Photostatistics
-    auto const noisePhoto = (amplitude > pedWidth) ? std::sqrt(amplitude * channelData.fcByPE()) : 0.f;
+    auto const noisePhotoSq = (amplitude > pedWidth) ? (amplitude * channelData.fcByPE()) : 0.f;
 
     //Total uncertainty from all sources
-    nnlsWork_.noiseTerms.coeffRef(iTS) = noiseADC * noiseADC + noisePhoto * noisePhoto + pedWidth * pedWidth;
+    nnlsWork_.noiseTerms.coeffRef(iTS) = noiseADC * noiseADC + noisePhotoSq + pedWidth * pedWidth;
+
+    nnlsWork_.pedVals.coeffRef(iTS) = pedWidth * pedWidth;
```

Comment:
it looks like the only way this is used is via sqrt of the value. It should be more computationally effective to just save the value `pedWidth` here and remove sqrt below
```suggestion
    nnlsWork_.pedVals.coeffRef(iTS) = pedWidth;
```